### ISSUE TITLE
Aboriginal Languages / Langues autochtones

### ISSUE THREAD

**Issue by GetdeGist** [wet-boew/wet-boew#7033]
I'm looking at http://wet-boew.github.io/wet-boew/index-en.html
Under "multilingual": 33 languages, but no sign of Aboriginal languages.

Je regarde http://wet-boew.github.io/wet-boew/index-fr.html
Sous "multilingue": 33 langues mais aucun signe des langue autochtones

**Comment by pjackson28** [comment]
WET is flexible enough to add support for additional languages as long as the translations for each of the strings is provided for those languages. To date, we have not received any translations for Aboriginal languages. Do you have a suggested source for the translations for the Aboriginal languages that should be supported?

**Comment by GetdeGist** [comment]
Translation: Good question.
I was mostly thinking about special characters - like Inuktitut (lang="iu") - http://www.gov.nu.ca/iu

**Comment by RobJohnston** [comment]
Here's a couple that may help:
* Inuktitut Living Dictionary: http://www.livingdictionary.com/
* Microsoft Language Portal: http://www.microsoft.com/Language/en-US/Search.aspx

But looks like the Microsoft site is Roman Inuktitut instead of Inuinnaqtun.  What you can do is use the Microsoft site, for example enter "search" and see that the UI element is "Qinirli", then use the Living Dictionary to see that "Qinirli" is ᓯᕿᓂᕐᓕᖅ in syllabic Inuktit, but the gov.nu.ca site uses ᕿᓂᕆᐊᓕᒃ... so I don't know how well this method is working.

**Comment by Bgolden67** [comment]
The requirement for Inuktitut support has been brought up a number of times over the past couple years to various working groups. I'm surprised to see @pjackson28 say there has been no requests for it. We (AANDC) have brought it up but never been approached with anything.

We've also brought the question up about how WET will handle the case where only a few pages need to be in the alternative language. Not all our content goes up in Inultitut but once in a while we have a need for the third language, but there has been no guidance on how to link to/from it template wise especially going forward with the new MWS

**Comment by pjackson28** [comment]
@Bgolden67 I didn't say there have been no requests for it, I said that nobody has offered to do the translations. It is very easy to add support as long as someone provides the translated strings. Would AANDC be willing to provide the translations for each of the strings? If so, we can provide the strings to translate.

As for how to handle a few pages in an alternate language, that isn't a WET question but instead a Canada.ca question (Content & IA spec). WET would just produce a template for whatever is decided for such scenarios.

**Comment by Bgolden67** [comment]
I understand.. Sure, I can see what we can do to get some translations. Send me the strings and I will find out what i can do to get them translated by our channels.

Correct.. I tend to lump WET in as everything, sorry about that.. but yes it has been brought up to multiple WGs and no answers have been given.

**Comment by pjackson28** [comment]
You can find all the strings here: https://docs.google.com/spreadsheets/d/1BmMrKN6Rtx-dwgPNEZD6AIAQdI4nNlyVVVCml0U594o/edit#gid=1. The strings to translate are on the "v4.0 translations" tab.

What we've done in the past is grant edit access to translators so they can put their translations right into the spreadsheet. So a new column would be added for each Inuit language (or other language) that we will be adding support for. WET then takes the strings from this spreadsheet (as part of the build process) and generates the necessary i18n files in WET.

**Comment by Bgolden67** [comment]
ok thanks.. I will find out what i need to do at this end and get the ball rolling!

**Comment by hsrudnicki** [comment]
Fine, it's further to the ball rolling, please update this issue's status whether this issue can be closed or otherwise?

**Comment by Bgolden67** [comment]
the link to the strings document is blocked from my desk, I will have to do this from home

**Comment by GetdeGist** [comment]
I'm happy to hear this ball is rolling. Thanks everyone. 

Thanks @Bgolden67 

Please keep us updated! :-)

**Comment by GetdeGist** [comment]
@Bgolden67  How is it moving along?

**Comment by Bgolden67** [comment]
I hope to have something in a week or so, sorry for the delays

**Comment by GetdeGist** [comment]
OK. Thanks!

**Comment by Bgolden67** [comment]
Should I add a new column in at the end or insert it in the correct alphabetical location?

**Comment by pjackson28** [comment]
@Bgolden67 It is currently ordered alphabetically by "Language name (English)" (with the exception of English and French which are moved to the front). For the sake of consistency it is suggested that you continue with the same alphabetical ordering.

**Comment by Bgolden67** [comment]
thanks!

**Comment by Bgolden67** [comment]
I'm waiting for the words and phrases to come back from translation, but How will we be handling the font requirement?
On the AANDC site we have offered the Pigiarniq font to download otherwise you probably can't see the Inuktitut characters.

**Comment by RobJohnston** [comment]
Isn't download a font a problem for Windows XP and before?  Hasn't Vista onwards added support for enough unicode for syllabics?  I've done nothing but can see http://www.gov.nu.ca/iu rendered with syllabics.

**Comment by Bgolden67** [comment]
Yes the support has been greatly improved, it's just something we've tried to make available previously because not all computers, especially in remote locations are up to date. Going forward I think it's going to be far less of a concern than in the past of course.

We also had instructions on the site on how to install for XP and Win7.

Just something i wanted to make sure was brought up to see if it was worth a concern still.

**Comment by GetdeGist** [comment]
The "font question" is not limited to Inuktitut only. 

- For example, in Chipewyan, you've got the "the glottal stop" – ʔ . 
- In old Huron-Wendat, Abenaki, Algonquin, you've got the "latin ou" - ȣ

- There might be other instances of non-Roman characters; I'm not an expert.

There's been a story on CBC about a person using the "glottal stop" in a name. This trend of reclaiming or re-owning traditional names, traditional spellings, etc - this is something that I think we have to accept, value and treasure. Even in the case of the "latin ou" which is being phased out (being replaced with ô or with w), I think it would be safest for us to make room or design a strategy for it - in case it appears in a name or in older documents, or in case it makes a come back for a reason, etc.

http://www.cbc.ca/radio/unreserved/tradition-authenticity-and-the-fight-for-indigenous-identity-1.3281731/what-s-in-a-name-why-three-people-are-fighting-for-their-traditional-names-1.3282113

https://en.wikipedia.org/wiki/Ou_(ligature)

**Comment by RobJohnston** [comment]
Yes, there are some unicode that I can't get... the last 9 in the range of [Unified Canadian Aboriginal Syllabics] (http://www.alanwood.net/unicode/unified_canadian_aboriginal_syllabics.html) and all of the [Unified Canadian Aboriginal Syllabics Extended] (http://www.alanwood.net/unicode/unified-canadian-aboriginal-syllabics-extended.html).  See what they should look like [here](http://www.unicode.org/charts/PDF/U1400.pdf) and [here](http://www.unicode.org/charts/PDF/U18B0.pdf).

Interesting that the glottal stop looks different - ᐞ.

**Comment by Bgolden67** [comment]
I have translations for inuktitut I will be adding this weekend, and I will also be submitting for the 4th language we rarely but on ocaison have to support -  <code>Inuinnaqtun</code> - once i get it back from translation

**Comment by GetdeGist** [comment]
That's good news, Brian. Thanks.

**Comment by hsrudnicki** [comment]
Based on the above, can be this issue closed or otherwise?
Either way, please update this issue's status, thank you

**Comment by GetdeGist** [comment]
1. Brian is working on translating  Inuinnaqtun.

2. Rob Johnston raised questions about Unicode - no one has replied yet specifically on this question.

3. I see that work has been done/is being done on Inuktitut and Inuinnaqtun but we have not discussed what to do/what should be done in respect to other indigenous languages. Should they be proactively added to ensure that the support is there whenever it is needed? We may want to build that support/capacity in anticipation of possible future needs - following the Truth and Reconciliation Commission, etc. The demands possibly may expand and increase beyond our current models, more requirements, more clients, more expectations for resources in indigenous languages, more need for opportunities to read/learn/use these languages, more people training and learning the languages, etc.

Conclusion: I think this issue is still active.

**Comment by Bgolden67** [comment]
Oh crap .. i forgot to update the Inuktitut in the file.. i will this week for sure

**Comment by hsrudnicki** [comment]
Does this remain still an active issue or otherwise? Please update the status of this again, whether this can be closed as resolved or otherwise, thank you

**Comment by Bgolden67** [comment]
still active

**Comment by GetdeGist** [comment]
Thanks Brian.

**Comment by RobJohnston** [comment]
Remember that unicode problem? Well, last week Google launched a new font called Noto.

> Today, Google’s open-source Noto font family provides a beautiful and consistent digital type for every symbol in the Unicode standard, covering more than 800 languages and 110,000 characters.

https://developers.googleblog.com/2016/10/an-open-source-font-system-for-everyone.html?m=1

**Comment by GetdeGist** [comment]
Thanks Rob. This is very interesting!

They list the following 11 Canadian Aboriginal Languages:

Chipewyan (Canadian Aboriginal script)
Cree
Northern East Cree
Southern East Cree
Inuktitut (Canadian Aboriginal script)
Moose Cree
Naskapi
Ojibwa (Canadian Aboriginal script)
Plains Cree
Slave (Canadian Aboriginal script)
Swampy Cree

**Comment by Bgolden67** [comment]
Can you grant me edit access to the file please? I sent the request using my gmail address

**Comment by GetdeGist** [comment]
Hi Brian.

What do I need to do or how can I grant you edit access? Where do I go?

**Comment by Bgolden67** [comment]
I'm not sure who the "owner" is of the google doc for the language translations file. I have the Inuktitut translations to add

**Comment by Bgolden67** [comment]
the file @pjackson28 mentions earlier in this thread

**Comment by GetdeGist** [comment]
Do you mean?

- https://docs.google.com/spreadsheets/d/1BmMrKN6Rtx-dwgPNEZD6AIAQdI4nNlyVVVCml0U594o/edit#gid=1

I don't own it.

I assume Paul ( @pjackson28 ) might know.

**Comment by Bgolden67** [comment]
yes that's the one

**Comment by GetdeGist** [comment]
I don't own it.

I assume Paul ( @pjackson28 ) might know.

**Comment by kodecount** [comment]
I think were suppose too edit this file https://github.com/wet-boew/wet-boew/blob/master/src/i18n/i18n.csv

**Comment by LaurentGoderre** [comment]
No, that files ia generated from the Google Spreadsheet

**Comment by pjackson28** [comment]
@Bgolden67 You should have edit access now.

**Comment by Bgolden67** [comment]
thanks @pjackson28

**Comment by Bgolden67** [comment]
column and translations have been added for Inuktitut

**Comment by GetdeGist** [comment]
Big thanks and big cheers to everyone. Special thanks to @Bgolden67 . And again thanks to everyone.

A small step for us, a bigger one for Inuktitut and for reconciliation! This is something real. It's not imaginary or abstract. It's real stuff!

Next Steps...
Is there more work to do, are there any projects ongoing? What's next?

**Comment by Bgolden67** [comment]
Glad to help :-)

The bigger issue i am facing is how the AEM CMS will handle these alternate languages, I know that is a discussion for another place and not this thread but it is a big concern we need to have the principal publisher tackle head on and fast.

**Comment by RobJohnston** [comment]
@GetdeGist You ask "what's next". Maybe now a PR to update the supported languages on this page:
http://wet-boew.github.io/wet-boew/index-en.html#multilingual

**Comment by LaurentGoderre** [comment]
There's another pr first to bring the update to the spreadsheet. I'll try to do this on Monday

**Comment by LaurentGoderre** [comment]
@GetdeGist thanks to @Bgolden67 version v4.0.23 will have Inuktitut!

**Comment by GetdeGist** [comment]
Thank you @LaurentGoderre ! That's awesome!

As @RobJohnston said, we will need to update the list of languages at http://wet-boew.github.io/wet-boew/index-en.html#multilingual 

How do we do that?

**Comment by GetdeGist** [comment]
@Bgolden67 also mentioned Inuinnaqtun. 

How is that other language going?

**Comment by RobJohnston** [comment]
@GetdeGist :  How about you do your first pull request?  Feel up to it?

The briefest of instructions are here: 
http://wet-boew.github.io/wet-boew/docs/start-en.html#develop

The files to be updated are here:
https://github.com/wet-boew/wet-boew/blob/master/site/pages/index-en.hbs
https://github.com/wet-boew/wet-boew/blob/master/site/pages/index-fr.hbs

**Comment by Bgolden67** [comment]
@GetdeGist I know I made the translation request, but i totally forgot about it because i don't recall it ever coming in. I will follow up and get the translations for Inuinnaqtun.

**Comment by GetdeGist** [comment]
@RobJohnston : I like challenges. I'm not a developer (I work in communications), I'm a beginner in GitHub and WET and all - but I will try my best to pull my first request. 

I've read the instructions but there are things I'm not familiar with - so I've asked Service Canada to train me a little bit on how things work.

**Comment by kodecount** [comment]
@GetdeGist I suggest [Atom](https://atom.io/) as a editor if you do not have one.

**Comment by GetdeGist** [comment]
Thanks Kody @kodecount  - I don't have an editor. I will probably need one I guess... 

I first need to know if I can download and install stuff on my computer or if I need approvals or permissions or whatnot. (Same with NodeJS - can I just download and install it myself right away?)

The instructions ( http://wet-boew.github.io/wet-boew/docs/start-en.html#develop ) also mention a "command line console". Another unknown for me. Humble beginnings for me...

**Comment by kodecount** [comment]
@GetdeGist Yes `git for windows` will install a bash command line terminal.

**Comment by kodecount** [comment]
@GetdeGist 

You'll need to `fork` the project on GitHub site (simplest way) before starting anything.

Then in the git bash terminal run these commands:
```
git config --global user.name "YOUR NAME"

git config --global user.email "YOUR EMAIL"

git clone https://github.com/ YOUR-USERNAME /wet-boew.git

cd wet-boew

npm install -g grunt grunt-cli yo bower

./script/setup

git remote add upstream https://github.com/wet-boew/wet-boew.git

git fetch upstream

git checkout -b YOUR-BRANCH-NAME
```

**Comment by kodecount** [comment]
@GetdeGist Hope that helps I found the WET documents to be out date a bit and confusing.

**Comment by GetdeGist** [comment]
It helps. (Thanks!) 
But I will need time to work on this one step at a time. :-)

**Comment by Bgolden67** [comment]
@pjackson28 Do you know who is leading the Content and IA these days so I can chase them down on how they plan to handle single page third/fourth languages as you mentioned early on in this thread it's not a WET decision just building the templates based on their requirements?

cheers
Brian

**Comment by pjackson28** [comment]
@Bgolden67 I suggest reaching out to @DK-TBS or @pcwsmith from Content & IA as either they are involved are or are connected to those involved with the work on the multiple-language approach.

**Comment by Bgolden67** [comment]
Thanks @pjackson28

**Comment by Bgolden67** [comment]
@DK-TBS and @pcwsmith Any guidance you can share?
